src/tx_broadcaster.rs

```diff
@@ -23,72 +25,90 @@ where
 	queue_receiver: Mutex<mpsc::Receiver<Vec<Transaction>>>,
 	esplora_client: EsploraClient,
 	logger: L,
+	channel_scheduler: Arc<Mutex<ChannelScheduler>>,
 }
 
 impl<L: Deref> TransactionBroadcaster<L>
 where
 	L::Target: Logger,
 {
-	pub(crate) fn new(esplora_client: EsploraClient, logger: L) -> Self {
+	pub(crate) fn new(
+		esplora_client: EsploraClient, logger: L, channel_scheduler: Arc<Mutex<ChannelScheduler>>,
+	) -> Self {
 		let (queue_sender, queue_receiver) = mpsc::channel(BCAST_PACKAGE_QUEUE_SIZE);
-		Self { queue_sender, queue_receiver: Mutex::new(queue_receiver), esplora_client, logger }
+		Self {
+			queue_sender,
+			queue_receiver: Mutex::new(queue_receiver),
+			esplora_client,
+			logger,
+			channel_scheduler,
+		}
 	}
 
 	pub(crate) async fn process_queue(&self) {
 		let mut receiver = self.queue_receiver.lock().await;
 		while let Some(next_package) = receiver.recv().await {
 			for tx in &next_package {
-				match self.esplora_client.broadcast(tx).await {
-					Ok(()) => {
-						log_trace!(self.logger, "Successfully broadcast transaction {}", tx.txid());
-					},
-					Err(e) => match e {
-						esplora_client::Error::Reqwest(_) => {
-							// Wait 500 ms and retry in case we get a `Reqwest` error (typically
-							// 429)
-							tokio::time::sleep(Duration::from_millis(500)).await;
-							log_error!(
-								self.logger,
-								"Sync failed due to HTTP connection error, retrying: {}",
-								e
-							);
-							match self.esplora_client.broadcast(tx).await {
-								Ok(()) => {
-									log_debug!(
-										self.logger,
-										"Successfully broadcast transaction {}",
-										tx.txid()
-									);
-								},
-								Err(e) => {
-									log_error!(
-										self.logger,
-										"Failed to broadcast transaction {}: {}",
-										tx.txid(),
-										e
-									);
-									log_trace!(
-										self.logger,
-										"Failed broadcast transaction bytes: {}",
-										log_bytes!(tx.encode())
-									);
-								},
-							}
-						},
-						_ => {
-							log_error!(
-								self.logger,
-								"Failed to broadcast transaction {}: {}",
-								tx.txid(),
-								e
-							);
+				let mut channel_scheduler = self.channel_scheduler.lock().await;
+				if channel_scheduler.is_funding_tx_created(tx.txid()) {
```

Comment:
so when the user calls `payjoin_open_channel` the channel is opened and stopped after `FundingGenerationReady` (https://github.com/lightningdevkit/ldk-node/pull/257/files#diff-14da7c350de5742eaa248de110fefeae9b4b639c0358e26024d6040ccf174a41R359) and the user gets back the bip21 they need to pay, nothing with payjoin until now. the channel now is already in `ChannelAccepted` phase and will be continued when the user pays the bip21, here we make sure we finish the communication with the counterparty node(ie get FundingSigned) before we return response(https://github.com/lightningdevkit/ldk-node/pull/257/files#diff-a5e41e49b8b7e8dd8ec88ed6b1c5ba0b530c2a186e37db14617223e34b11ca31R174) this is so we dont lock the funds in case we return a response the the payjoin sender without getting FundingSigned from the node we want to open channel with.